### ISSUE TITLE
RCS1170 suppress when property has DataMember attribute

### ISSUE THREAD

**Issue by silvioboehme** [JosefPihrt/Roslynator#70]
Hi!

I have a [DataContract] for a immutable class where all properties are public get and private set.
All Properies are decorated with [DataMember].

RCS1170 suggest, that i should remove private set. This leads to serailization/deserialization Errors.

`[DataContract]`
`public sealed class User`
`{`
`[DataMember]`
`        public Guid Id { get; private set; } = Guid.NewGuid();`

`        [DataMember]`
`        public string Name { get; private set; }`
`}`

RCS1170 fires for property Id, because this isn't set internaly, but needs to serialized/deserialized properly.
RCS1170 isn't fired for property Name, because it's set within the class.

I know it's a rare case. Also changing Id to readonly and use a backing field for Id leads to RCS1169 fired for the backing field, (field should be readonly) which is also not possible.

Thanks
Silvio

**Comment by JosefPihrt** [comment]
Thanks for the report @silvioboehme !

**Comment by jnyrup** [comment]
This is also the case for other contract based serializers, e.g. protobuf-net's [ProtoMember] attribute.

**Comment by GothicSecret** [comment]
same happened to me, we accidently removed property setter marked with [JsonProperty]

**Comment by JosefPihrt** [comment]
@GothicSecret fixed with 1545a7d3ebe21b15b6ccdfbaec0deb71a74333e8

**Comment by taylorchasewhite** [comment]
I noticed this happened in Blazor code too, it should be suppressed if there is a [`Microsoft.AspNetCore.Components.InjectAttribute`] attribute on the property:

```
	[Inject]
	private DbContexts Contexts { get; set; } = null!;
```
